test/e2e/specs/editor/various/copy-cut-paste.spec.js

```diff
@@ -478,6 +478,23 @@ test.describe( 'Copy/cut/paste', () => {
 		).toBe( 'axyb' );
 	} );
 
+	test( 'should paste formatted text into focusable element', async ( {
+		page,
+		pageUtils,
+		editor,
+	} ) => {
+		await editor.canvas
+			.locator( 'role=button[name="Add default block"i]' )
+			.click();
+		await page.keyboard.type( 'test' );
+		await pageUtils.pressKeys( 'primary+a' );
+		await pageUtils.pressKeys( 'primary+b' );
+		await pageUtils.pressKeys( 'primary+c' );
+		await page.keyboard.press( 'ArrowRight' );
```

Comment:
I think you'll need to make sure the caret is _within_ the bold format?

Reply:
Strangely, the problem does not occur _inside_ formatted text, but only occurs when you move the caret _outside_ the formatted text and paste:

https://github.com/user-attachments/assets/9891efaf-cb5c-438c-a3fb-46a2c3f86a0f

Reply:
To trigger the link behaviour, you'd need to _select_ text within the format?

Reply:
I'd for example select a letter within the text format (to make sure the test is stable and selection boundaries are within the format), then paste the link on the uncollapsed selection

Reply:
I tried copying letters within the formatted text and the e2e test still passed on trunk 😅 Of course, the test is expected to fail on trunk.

https://github.com/user-attachments/assets/fb9bc6b3-ceca-4724-8a4c-bf05c80bfd92

Reply:
But you're not pasting a link?

Reply:
Ah, I see what you mean.

#63434 reports two issues:

1. If the text is formatted, highlighting and adding the link via paste doesn't work
2. If the copied text contains formatted text at the end, pasting the copied text outside the formatted text removes the formatting

This PR solves both problems, but the E2E test I added is for the second problem. Does this mean I should write a test for the first problem?

Reply:
I didn't see the second problem, I only knew about the first problem from the PR description. Sure, if it's easier to write a PR for.

Reply:
There may be slight differences with how paste is emulated that make it pass.

Reply:
I updated the e2e test to paste the link into the formatted text. But for some reason, this test still passes on trunk as well 😅